### ISSUE TITLE
Incoming call notification doesn't close, covers video call

### ISSUE THREAD

**Issue by Azelphur** [BelledonneCommunications/linphone-android#1486]
**Describe the bug**
When you receive an incoming call, the notification to answer the call appears as expected, but once you have answered the call, the notification does not close, covers the UI, and is impossible to dismiss

**To Reproduce**
1. Receive a call
2. Answer the call

**Expected behavior**
Notification should be closed once you have answered the call

**Please complete the following information**
 - Device: Oneplus 6
 - OS: Android 10
 - Version of the App 4.5.2
 - Version of the SDK 5.0.11

**Screenshots**
https://www.youtube.com/watch?v=6oCkMTQ0fHs

**Comment by Viish** [comment]
When call is answered, there is another notification that replaces the incoming call one, but from your video that didn't happen.
Can you enable debug logs in advanced settings, restart the app, reproduce the issue, upload logs and copy/paste the link here please?

**Comment by Azelphur** [comment]
Here's the log https://www.linphone.org:444//tmp/6129fdec1dfd9_d128e6235ae5cf38040d.gz

Thanks :)

**Comment by Viish** [comment]
That's weird, everything is fine in the logs:
```
2021-08-28 10:12:00:996 [org.linphone/linphone-android] MESSAGE [Notifications Manager] Call state changed [IncomingReceived]
2021-08-28 10:12:00:999 [org.linphone/linphone-android] MESSAGE [Notifications Manager] Notifying incoming call notification
2021-08-28 10:12:00:999 [org.linphone/linphone-android] MESSAGE [Notifications Manager] Notifying 12
[...]
2021-08-28 10:12:02:742 [org.linphone/linphone-android] MESSAGE [Notifications Manager] Call state changed [Connected]
2021-08-28 10:12:02:748 [org.linphone/linphone-android] MESSAGE [Notifications Manager] Notifying 12
```

What version of Android does your device runs ?

**Comment by Azelphur** [comment]
It's a Oneplus 6 running Android 10. I also experience the same issue on a Oneplus 7 Pro. Will test on my friends Galaxy Z Fold 3 and report back.

**Comment by Azelphur** [comment]
Doesn't seem to happen on my friends Galaxy Z Fold, perhaps it's an issue specific to Oneplus phones.

**Comment by Viish** [comment]
> Doesn't seem to happen on my friends Galaxy Z Fold, perhaps it's an issue specific to Oneplus phones.

That's interesting. 
Sadly we don't have a OnePlus device at work, I'll see if I can borrow one from someone to test on my side and find a fix if I can reproduce it.

**Comment by Azelphur** [comment]
I grabbed an adb logcat in the hopes it would be useful

```
# Placing call
08-28 10:59:29.105 26394 26394 E Linphone: Unable to find media attribute capabilities for stream 0
08-28 10:59:29.106 26394 26394 E Linphone: Unable to find media transport capabilities for stream 0
08-28 10:59:29.108 26394 26394 E Linphone: Unable to find media attribute capabilities for stream 1
08-28 10:59:29.109 26394 26394 E Linphone: Unable to find media transport capabilities for stream 1
08-28 10:59:29.110 26394 26394 E Linphone: Unable to find configurations for stream 0
08-28 10:59:29.111 26394 26394 E Linphone: Unable to find media attribute capabilities for stream 0
08-28 10:59:29.112 26394 26394 E Linphone: Unable to find media transport capabilities for stream 0
08-28 10:59:29.113 26394 26394 E Linphone: Unable to find configurations for stream 1
08-28 10:59:29.114 26394 26394 E Linphone: Unable to find media attribute capabilities for stream 1
08-28 10:59:29.114 26394 26394 E Linphone: Unable to find media transport capabilities for stream 1
08-28 10:59:29.121 26394 26394 W Linphone: Fail to increase socket's recv buffer size with SO_RCVBUFFORCE: Operation not permitted.
08-28 10:59:29.121 26394 26394 W Linphone: Fail to increase socket's send buffer size with SO_SNDBUFFORCE: Operation not permitted.
08-28 10:59:29.121 26394 26394 W Linphone: Fail to increase socket's recv buffer size with SO_RCVBUFFORCE: Operation not permitted.
08-28 10:59:29.122 26394 26394 W Linphone: Fail to increase socket's recv buffer size with SO_RCVBUFFORCE: Operation not permitted.
08-28 10:59:29.122 26394 26394 W Linphone: Fail to increase socket's send buffer size with SO_SNDBUFFORCE: Operation not permitted.
08-28 10:59:29.122 26394 26394 W Linphone: Fail to increase socket's recv buffer size with SO_RCVBUFFORCE: Operation not permitted.
08-28 10:59:29.136 26394 26394 E Linphone: [Notifications Manager] There is already a foreground service notification
08-28 10:59:29.137  1320  4916 D VibratorService: vibrate from 'org.linphone', usageHint = 6
08-28 10:59:29.143  1320  4916 I MediaFocusControl: requestAudioFocus() from uid/pid 10496/26394 clientId=android.media.AudioManager@66961d1org.linphone.core.tools.audio.AudioHelper@f528436 callingPack=org.linphone req=4 flags=0x0 sdk=30
08-28 10:59:29.190  1320  4920 I AS.AudioService: releasePlayer: 535 currentPackageName:org.linphone
08-28 10:59:29.239  1320  4920 I AS.AudioService: playerEvent: 543 event: 2 isMusicActive:false device:2 currentPackageName:org.linphone
08-28 10:59:29.240  1320  4920 I OpAudioMonitor: MDM,start put pk:org.linphone startTime:1630144769
08-28 10:59:29.266  4340  4340 D NotificationListener: onNotificationPosted# sbn: 151289240, package = org.linphone
08-28 10:59:29.273 21856 21974 D BaseStreamManager: Promised set @ r#5836063 for (org.linphone,tag=null,id=8,notifKey=0|org.linphone|8|null|10496)
08-28 10:59:29.276 21856 21980 D DynamicBucketRankerImpl: StreamItemIdAndRevision{notifKey=0|org.linphone|8|null|10496, packageName=org.linphone, id=8, revision=5836063, originalRevision=5836063, updateCount=0} in bucket #2 (UnreadRecentInterruptive)
08-28 10:59:29.279 21856 21974 I CBridgedItemsController: not bridging notification from org.linphone: notification is tagged as ongoing
08-28 10:59:29.751  1320 27206 D VibratorService: vibrate from 'org.linphone', usageHint = 5
# Pressing answer
08-28 10:59:33.375 26394 26394 W Linphone: [File Utils] External storage is mounted
08-28 10:59:33.405  1320  4925 D OemSceneCallBlock: isCallBlockedWithUidIntent { flg=0x10020000 cmp=org.linphone/.activities.call.CallActivity }, ResolveInfo{8ca1df8 org.linphone/.activities.call.CallActivity m=0x0}, false
08-28 10:59:33.406  1320  4925 I ActivityTaskManager: START u0 {flg=0x10020000 cmp=org.linphone/.activities.call.CallActivity} from uid 10496 pid 26394
08-28 10:59:33.414  1320  4925 I MediaFocusControl: abandonAudioFocus() from uid/pid 10496/26394 clientId=android.media.AudioManager@66961d1org.linphone.core.tools.audio.AudioHelper@f528436
08-28 10:59:33.421  1320  4925 I AS.AudioService: releasePlayer: 543 currentPackageName:org.linphone
08-28 10:59:33.508 26394 26394 W Linphone: Advanced adaptive rate control requested but avpf-tmmbr is not activated in this stream. Reverting to basic rate control instead
08-28 10:59:33.523   900  8780 D AudioFlinger: opPackageName , checkedOpPackageName org.linphone
08-28 10:59:33.523   900  8780 I AudioFlinger: package name:org.linphone
08-28 10:59:33.670 26394 26394 W Linphone: AcousticEchoCanceler already enabled
08-28 10:59:33.670 26394 26394 I AudioManager: In stopBluetoothSco(), calling application: org.linphone
08-28 10:59:33.671 26394 26394 I AudioManager: In setBluetoothScoOn(), on: false, calling application: org.linphone
08-28 10:59:33.714 26394 26394 I AudioManager: In stopBluetoothSco(), calling application: org.linphone
08-28 10:59:33.714 26394 26394 I AudioManager: In setBluetoothScoOn(), on: false, calling application: org.linphone
08-28 10:59:33.715  1320  5018 I AS.AudioService: playerEvent: 559 event: 2 isMusicActive:false device:2 currentPackageName:org.linphone
08-28 10:59:33.715  1320  1320 D AS.AudioService: adjustSuggestedStreamVolume() stream=0, flags=0, caller=org.linphone, volControlStream=-1, userSelect=false
08-28 10:59:33.715  1320  1320 D AS.AudioService: adjustSuggestedStreamVolume() stream=0, flags=0, caller=org.linphone, volControlStream=-1, userSelect=false
08-28 10:59:33.715  1320  5018 I OpAudioMonitor: MDM,start put pk:org.linphone startTime:1630144773
08-28 10:59:33.717  1320  1320 D AS.AudioService: adjustStreamVolume() stream=0, dir=-1, flags=0, caller=org.linphone
08-28 10:59:33.718  1320  1320 D AS.AudioService: adjustSuggestedStreamVolume() stream=0, flags=0, caller=org.linphone, volControlStream=-1, userSelect=false
08-28 10:59:33.718  1320  1320 D AS.AudioService: adjustSuggestedStreamVolume() stream=0, flags=0, caller=org.linphone, volControlStream=-1, userSelect=false
08-28 10:59:33.721 26394 26394 W Linphone: Advanced adaptive rate control requested but avpf-tmmbr is not activated in this stream. Reverting to basic rate control instead
08-28 10:59:33.731 26394 26394 W Linphone: No 'default' video preset has been found
08-28 10:59:33.731 26394 26394 W Linphone: [Camera2 Capture] Capture was already stopped, ignoring...
08-28 10:59:33.733 26394 26394 E Linphone: [Camera2 Capture] TextureView wasn't set (was core.setNativePreviewWindowId() called?)
08-28 10:59:33.760 26394 26394 E Linphone: no such method on filter MSMediaCodecH264Decoder, fid=16389 method index=7
08-28 10:59:33.760 26394 26394 E Linphone: no such method on filter MSAndroidTextureDisplay, fid=16387 method index=1
08-28 10:59:33.760 26394 26394 E Linphone: no such method on filter MSAndroidTextureDisplay, fid=16387 method index=0
08-28 10:59:33.765 26394 27222 I ACodec  : setupAVCEncoderParameters: Get package name: org.linphone
08-28 10:59:33.802  1320  4916 I AS.AudioService: releasePlayer: 559 currentPackageName:org.linphone
08-28 10:59:33.804  1320  1320 D AS.AudioService: adjustStreamVolume() stream=0, dir=1, flags=0, caller=org.linphone
08-28 10:59:33.913  1320  1395 W ActivityTaskManager: Activity top resumed state loss timeout for ActivityRecord{55dc221 u0 org.linphone/.activities.main.MainActivity t27515}
08-28 10:59:33.913  1320  1395 W ActivityTaskManager: Activity pause timeout for ActivityRecord{55dc221 u0 org.linphone/.activities.main.MainActivity t27515}
08-28 10:59:33.915  1320  1395 D OpQuickReply: setQuickReplyResumed focusedApp AppWindowToken{e057837 token=Token{bb99136 ActivityRecord{90082d1 u0 org.linphone/.activities.call.CallActivity t27515}}} pkgName org.linphone
08-28 10:59:33.920  1320  1395 D OpRestartProcessManager: updateSelf :  org.linphone, size : 30
08-28 10:59:34.038 26394 26394 I AudioManager: In stopBluetoothSco(), calling application: org.linphone
08-28 10:59:34.038 26394 26394 I AudioManager: In setBluetoothScoOn(), on: false, calling application: org.linphone
08-28 10:59:34.039  1320  1320 D AS.AudioService: adjustSuggestedStreamVolume() stream=0, flags=0, caller=org.linphone, volControlStream=-1, userSelect=false
08-28 10:59:34.039  1320  1320 D AS.AudioService: adjustSuggestedStreamVolume() stream=0, flags=0, caller=org.linphone, volControlStream=-1, userSelect=false
08-28 10:59:34.040  1320  5018 I AS.AudioService: playerEvent: 567 event: 2 isMusicActive:false device:2 currentPackageName:org.linphone
08-28 10:59:34.040  1320  2489 I MediaFocusControl: requestAudioFocus() from uid/pid 10496/26394 clientId=android.media.AudioManager@66961d1org.linphone.core.tools.audio.AudioHelper@f528436 callingPack=org.linphone req=2 flags=0x0 sdk=30
08-28 10:59:34.041  1320  5018 I OpAudioMonitor: MDM,start put pk:org.linphone startTime:1630144774
08-28 10:59:34.041 26394 26394 I AudioManager: In setMode(), mode: 3, calling application: org.linphone
08-28 10:59:34.042  1320  2489 I AS.AudioService: setMode(mode = 3, callingPackage = org.linphone, Process ID: 26394
08-28 10:59:34.042  1320  2489 V AS.AudioService: setMode(mode=3, callingPackage=org.linphone)
08-28 10:59:34.042  1320  2489 V AS.AudioService: setModeInt(mode=3, pid=26394, caller=org.linphone)
08-28 10:59:34.042   900  1036 D APM_AudioPolicyManager: setAudioPolicyParameters(): io 0, keyvalue setModeClient=org.linphone
08-28 10:59:34.069  1320  1320 D AS.AudioService: adjustStreamVolume() stream=0, dir=-1, flags=0, caller=org.linphone
08-28 10:59:34.083 26394 26394 D OnePlusJankManager:  Chor uploadMDM JANK_TYPE_ONCE mViewTitle = org.linphone/org.linphone.activities.main.MainActivity--- jank level = 3
08-28 10:59:34.084  1320  1408 I OPDiagnose: OPDiagnoseManager writeDiagData type = 30, issue_desc = pkg:org.linphone-view:org.linphone/org.linphone.activities.main.MainActivity-jkType:1-level:3
08-28 10:59:34.084  1320  1408 I         : BpOPDiagnoseServer writeDiagData type = 30, issue_desc = pkg:org.linphone-view:org.linphone/org.linphone.activities.main.MainActivity-jkType:1-level:3
08-28 10:59:34.084   867   918 I         : BnOPDiagnoseServer WRITE_DIAG_DATA type = 30, issue_desc = pkg:org.linphone-view:org.linphone/org.linphone.activities.main.MainActivity-jkType:1-level:3
08-28 10:59:34.084   867   918 I OPDiagnoseService: writeDiagData: type = 30, issue_desc = pkg:org.linphone-view:org.linphone/org.linphone.activities.main.MainActivity-jkType:1-level:3, time_now = 2021-08-28 10:59:34
08-28 10:59:34.085  1320  1320 D AS.AudioService: adjustSuggestedStreamVolume() stream=0, flags=0, caller=org.linphone, volControlStream=-1, userSelect=false
08-28 10:59:34.085  1320  1320 D AS.AudioService: adjustSuggestedStreamVolume() stream=0, flags=0, caller=org.linphone, volControlStream=-1, userSelect=false
08-28 10:59:34.088  1320  4925 D ActivityTaskManager: setRequestedOrientation() to -1 for ActivityRecord{90082d1 u0 org.linphone/.activities.call.CallActivity t27515} from pid=26394, uid=10496
08-28 10:59:34.089  1320  1320 D AS.AudioService: adjustStreamVolume() stream=0, dir=1, flags=0, caller=org.linphone
08-28 10:59:34.170 26394 27232 W Linphone: [ogl_functions] Function : Fail to load plugin libGLEW.so: dlopen failed: library "libGLEW.so" not found
08-28 10:59:34.178   795   795 D CHIUSECASE: [initPackageName][2637] "[OP_EXT] org.linphone"
08-28 10:59:34.179  1320  1408 I ActivityTaskManager: Displayed org.linphone/.activities.call.CallActivity: +771ms
08-28 10:59:34.181   795   795 I         : LoadOPPackageName: [OP_EXT] org.linphone
08-28 10:59:34.183   795   795 I         : LoadOPPackageName: [OP_EXT] org.linphone
08-28 10:59:34.531 26394 27213 W Linphone: MSAudio MSTicker: We are late of 219 miliseconds.
08-28 10:59:34.651 26394 27233 E Linphone: MediaCodecEncoder: I-frame requested to MediaCodec
08-28 10:59:34.781 26394 27233 W Linphone: MSVideo MSTicker: We are late of 172 miliseconds.
08-28 10:59:35.124 26394 26394 W Linphone: linphone_call_stats_get_receiver_loss_rate(): there is no RTCP packet received.
# Pressing answer during call
08-28 10:59:35.937 26394 26394 W Linphone: linphone_call_stats_get_receiver_loss_rate(): there is no RTCP packet received.
08-28 10:59:36.551 26394 27233 E Linphone: MediaCodecEncoder: I-frame requested to MediaCodec
08-28 10:59:37.467 26394 26394 W Linphone: [File Utils] External storage is mounted
08-28 10:59:37.470 26394 26394 E Linphone: checkForAcceptation() CallSession [0x7bd4624598] is in state [LinphoneCallStreamsRunning], operation not permitted
08-28 10:59:38.202 26394 26394 W Linphone: linphone_call_stats_get_receiver_loss_rate(): there is no RTCP packet received.
08-28 10:59:38.583 26394 27233 E Linphone: MediaCodecEncoder: I-frame requested to MediaCodec
08-28 10:59:38.862 26394 27213 W Linphone: Flow controlled bufferizer owned by [MSAAudioPlayer] of max 40 ms is filled with 210 ms at the end of the 5000 ms interval, asking to drop 85 ms
08-28 10:59:39.653 26394 27233 E Linphone: MediaCodecEncoder: I-frame requested to MediaCodec
# Pressing hangup
08-28 10:59:42.236 26394 26394 I AudioManager: In stopBluetoothSco(), calling application: org.linphone
08-28 10:59:42.237 26394 26394 I AudioManager: In setBluetoothScoOn(), on: false, calling application: org.linphone
08-28 10:59:42.266 26394 26394 I AudioManager: In stopBluetoothSco(), calling application: org.linphone
08-28 10:59:42.266  1320  5018 I AS.AudioService: playerEvent: 567 event: 4 isMusicActive:false device:1 currentPackageName:org.linphone
08-28 10:59:42.274 26394 26394 I AudioManager: In setBluetoothScoOn(), on: false, calling application: org.linphone
08-28 10:59:42.276  1320  5018 I AS.AudioService: releasePlayer: 567 currentPackageName:org.linphone
08-28 10:59:42.357 26394 26394 W Linphone: ms_yuv_buf_allocator_free(): leaving 1 mblk_t still ref'd, possible leak.
08-28 10:59:42.587 26394 26394 E Linphone: [Camera2 Capture] Couldn't abort captures on session, error is ACAMERA_ERROR_SESSION_CLOSED
08-28 10:59:42.587 26394 26394 E Linphone: [Camera2 Capture] Couldn't stop repeating session, error is ACAMERA_ERROR_SESSION_CLOSED
08-28 10:59:42.587 26394 26394 E BufferQueueConsumer: disconnect by : org.linphone my_pid = 26394
08-28 10:59:42.587  1106  2140 I Camera2ClientBase: Closed Camera 1. Client was: org.linphone (PID 26394, UID 10496)
08-28 10:59:42.607  1320  4925 I MediaFocusControl: abandonAudioFocus() from uid/pid 10496/26394 clientId=android.media.AudioManager@66961d1org.linphone.core.tools.audio.AudioHelper@f528436
08-28 10:59:42.607 26394 26394 I AudioManager: In setMode(), mode: 0, calling application: org.linphone
08-28 10:59:42.607  1320  4925 I AS.AudioService: setMode(mode = 0, callingPackage = org.linphone, Process ID: 26394
08-28 10:59:42.607  1320  4925 V AS.AudioService: setMode(mode=0, callingPackage=org.linphone)
08-28 10:59:42.607  1320  4925 V AS.AudioService: setModeInt(mode=0, pid=26394, caller=org.linphone)
08-28 10:59:42.607   900  8462 D APM_AudioPolicyManager: setAudioPolicyParameters(): io 0, keyvalue setModeClient=org.linphone
08-28 10:59:42.693  4340  4340 D NotificationListener: onNotificationRemoved# sbn: 12450289, package = org.linphone
08-28 10:59:42.694 21856 21974 D BaseStreamManager: Promised remove @ r#5836064 for StreamItemIdAndRevision{notifKey=0|org.linphone|8|null|10496, packageName=org.linphone, id=8, revision=0, originalRevision=0, updateCount=0} (@5836063): SBN removed: StreamItemIdAndRevision{notifKey=0|org.linphone|8|null|10496, packageName=org.linphone, id=8, revision=0, originalRevision=0, updateCount=0}
08-28 10:59:42.694 21856 21980 D StreamDatabase: remove. itemId: StreamItemIdAndRevision{notifKey=0|org.linphone|8|null|10496, packageName=org.linphone, id=8, revision=5836063, originalRevision=5836063, updateCount=0}
08-28 10:59:43.020 26394 26394 I AudioManager: In stopBluetoothSco(), calling application: org.linphone
08-28 10:59:43.020 26394 26394 I AudioManager: In setBluetoothScoOn(), on: false, calling application: org.linphone
08-28 10:59:43.022  1320  1320 D AS.AudioService: adjustSuggestedStreamVolume() stream=0, flags=0, caller=org.linphone, volControlStream=-1, userSelect=false
08-28 10:59:43.022  1320  1320 D AS.AudioService: adjustSuggestedStreamVolume() stream=0, flags=0, caller=org.linphone, volControlStream=-1, userSelect=false
08-28 10:59:43.022  1320  3992 I AS.AudioService: playerEvent: 575 event: 2 isMusicActive:false device:2 currentPackageName:org.linphone
08-28 10:59:43.024  1320  3992 I OpAudioMonitor: MDM,start put pk:org.linphone startTime:1630144783
08-28 10:59:43.025  1320  1320 D AS.AudioService: adjustStreamVolume() stream=0, dir=-1, flags=0, caller=org.linphone
08-28 10:59:43.027  1320  4925 W ActivityTaskManager: Duplicate finish request for ActivityRecord{90082d1 u0 org.linphone/.activities.call.CallActivity t27515 f}
08-28 10:59:43.030  1320  4925 D OpQuickReply: setQuickReplyResumed focusedApp AppWindowToken{8fa9007 token=Token{97e646 ActivityRecord{55dc221 u0 org.linphone/.activities.main.MainActivity t27515}}} pkgName org.linphone
08-28 10:59:43.034  1320  1320 D AS.AudioService: adjustSuggestedStreamVolume() stream=0, flags=0, caller=org.linphone, volControlStream=-1, userSelect=false
08-28 10:59:43.034  1320  1320 D AS.AudioService: adjustSuggestedStreamVolume() stream=0, flags=0, caller=org.linphone, volControlStream=-1, userSelect=false
08-28 10:59:43.035 26394 26394 D OnePlusJankManager:  Chor uploadMDM JANK_TYPE_ONCE mViewTitle = org.linphone/org.linphone.activities.call.CallActivity--- jank level = 2
08-28 10:59:43.035  1320  1395 D OpRestartProcessManager: updateSelf :  org.linphone, size : 30
08-28 10:59:43.040  1320  1320 D AS.AudioService: adjustStreamVolume() stream=0, dir=1, flags=0, caller=org.linphone
08-28 10:59:43.093 26394 26394 W Linphone: [Platform Helper] Found an existing video TextureView, let's destroy it first
08-28 10:59:43.093 26394 26394 W Linphone: [Platform Helper] Found an existing preview TextureView, let's destroy it first
08-28 10:59:43.096 26394 26394 E BufferQueueConsumer: disconnect by : org.linphone my_pid = 26394
08-28 10:59:43.096 26394 26394 E BufferQueueConsumer: disconnect by : org.linphone my_pid = 26394
08-28 10:59:44.831 26394 26394 I AudioManager: In stopBluetoothSco(), calling application: org.linphone
08-28 10:59:44.832 26394 26394 I AudioManager: In setBluetoothScoOn(), on: false, calling application: org.linphone
08-28 10:59:44.833  1320  2489 I AS.AudioService: playerEvent: 575 event: 4 isMusicActive:false device:2 currentPackageName:org.linphone
08-28 10:59:44.834  1320  2489 I AS.AudioService: releasePlayer: 575 currentPackageName:org.linphone
```

**Comment by Azelphur** [comment]
Found the root cause. This issue occurs when the "Linphone service notification" is hidden in the app settings.

![Screenshot](https://i.imgur.com/01ZkFbq.png)

**Comment by Viish** [comment]
Great, thanks!
I'll be able to find a workaround for that.

**Comment by Viish** [comment]
I've merged the fix in master branch commit 46b9478b, thanks again for reporting the issue :)
